### PR TITLE
Changed group: hosts to allow inline variable assignment

### PR DESCRIPTION
**Pull request by jamessewell**
Hi,

Change to allow the following syntax:

```
- group: <groupname>
  vars:
  - groupvar: value
  hosts:
  - myhost1
  - host: myhost2
    vars:
      myvcr: vclue
  groups:
  - subgroup1
  - subgroup2
```

Standard hosts and dict hosts can be mixed and matched.

### CODE REVIEW

File: yaml-hosts.py [hunk 1]
```diff
@@ -189,6 +217,7 @@ def parse_yaml(yaml_hosts):
                 ungrouped.add_host(host)
 
         elif 'host' in entry:
+            print entry
```
Comment (jhoekx) [inline yaml-hosts.py]: I think we do not really want this print here, no?

File: yaml-hosts.py [hunk 2]
```diff
@@ -214,16 +243,21 @@ def parse_yaml(yaml_hosts):
...
 
+    for group in groups:
+      group.add_subgroup(all_hosts)
+
     return groups, all_hosts
 
+base_dir = os.path.dirname(os.path.realpath(__file__))
+
 parser = OptionParser()
+parser.add_option('-f', '--file', default=os.path.join(base_dir,"hosts.yaml"), dest="yaml_file")
```
Comment (jhoekx) [inline yaml-hosts.py]: Just for consistency: add a space before the `"hosts.yaml"`.

**Review by jhoekx — CHANGES_REQUESTED** [review]
Looks good, thanks. A few details that should probably be fixed though.

**Comment by jamessewell** [comment]
Hi,

Thanks!

So after doing some work I noticed that the format of your --list output
was ... strange.

Maybe this was written for a really old Ansible?

Anyway I'm halfway to supporting the output as defined in the developing
dynamic inventories page, just have to get the _meta stuff working and try
to work out how all is meant to work.

Cheers,
James Sewell

File: yaml-hosts.py [hunk 3]
```diff
@@ -136,111 +192,139 @@ def parse_vars(vars, obj):
...
-
-    return groups, all_hosts
+    for host in globals.all_hosts.get_hosts():
+        if len(host.groups) <= 1:
+            globals.ungrouped.add_host(host)
+
+base_dir = os.path.dirname(os.path.realpath(__file__))
 
 parser = OptionParser()
+parser.add_option('-f', '--file', default=os.path.join(base_dir, "hosts.yaml"), dest="yaml_file")
```
Comment (jhoekx) [inline yaml-hosts.py]: This default extension should be `.yml`

File: yaml-hosts.py [hunk 4]
```diff
@@ -136,111 +192,139 @@ def parse_vars(vars, obj):
...
+            group.add_parent(parent)
+
+    return group
+
+def parse_host(entry):
+    ### a host is either a dict or a single line definition
+    if type(entry) in [str, unicode]:
+        for test_host in globals.all_hosts.get_hosts():
+            if test_host.name == entry:
+                break
```
Comment (jhoekx) [inline yaml-hosts.py]: I think that this branch should also set the host variable.

**Review by jhoekx — CHANGES_REQUESTED** [review]
A few small thing I noticed when testing. With these fixed it was working in my example.

**Comment by jhoekx** [comment]
This was actually part of Ansible proper from version (I'm lying, but you get the idea) 0.3 to 0.6. It was never updated with the 'new' _meta syntax.

**Comment by jamessewell** [comment]
Ha, I see.

With these new changes it's really helping us to manage our inventory, hopefully it can make it back in!

I've done a bit of cleanup and added a label: tag for a group, which will set a variable with the groupname:

so 

```
group: james
label: name
```

would yield:

```
"james": {
        "children": [],
        "hosts": [],
        "vars": {
            "name": "james"
        }
```

**Comment by jhoekx** [comment]
Could you ping me when you're done making changes? Then I'll review and merge.

**Comment by jamessewell** [comment]
It's fine now.

I'm not 100% sold on the environment variable logic, but it's what we want
in production.

I'm also not sure on the merge vars behaviour, but once again it's what I
want for us.

Apart from that I'm happy. The import_vars stuff was working, but it seemed
too involved.

Cheers,
James Sewell

**Review by jhoekx — APPROVED** [review]

**Comment by jhoekx** [comment]
Merged. Thanks for bringing the yaml inventory in the modern age of ansible!

**Comment by jamessewell** [comment]
Hi,

Hey I just realised that my email address hasn't been pulling through - I
don't suppose you could update the entries?

If you look in my fork there is a script which does it which I committed
briefly - try.file.

Cheers,
James

**Comment by jhoekx** [comment]
Ok, changed. I also updated the name in the commits.

**Pull request merged**